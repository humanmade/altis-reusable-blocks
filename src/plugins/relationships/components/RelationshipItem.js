import PropTypes from 'prop-types';

import { PanelRow } from '@wordpress/components';
import { __ } from '@wordpress/i18n';

import { getEditPostUrl } from '../../../utils/post';

const statusMap = {
	draft: __( '(Draft)', 'newspress' ),
	pending: __( '(Pending)', 'newspress' ),
};

const RelationshipItem = ( { id, status, title } ) => (
	<PanelRow>
		<a target="_blank" rel="noopener noreferrer" href={ getEditPostUrl( id ) }>
			{ `#${ id } - ${ ( title.rendered || __( '(No Title)', 'altis-reusable-blocks' ) ) }` }
		</a>
		{ statusMap[ status ] || null }
	</PanelRow>
);

RelationshipItem.propTypes = {
	id: PropTypes.number.isRequired,
	status: PropTypes.string.isRequired,
	title: PropTypes.shape( {
		rendered: PropTypes.string,
	} ),
};

RelationshipItem.defaultProps = {
	title: {},
};

export default RelationshipItem;
