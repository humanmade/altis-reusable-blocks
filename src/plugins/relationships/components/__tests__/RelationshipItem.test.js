import { shallow } from 'enzyme';

import RelationshipItem from '../RelationshipItem';

jest.mock( '../../../../utils/post', () => ( {
	getEditPostUrl: ( postId ) => `https://example.com/edit/${ postId }`,
} ) );

jest.mock( '@wordpress/components', () => mockComponents( [
	'PanelRow',
] ) );

describe( '<RelationshipItem/>', () => {

	test( 'renders published post as expected', () => {
		const title = {
			rendered: 'Published Post',
		};

		const result = shallow(
			<RelationshipItem id={ 42 } status="publish" title={ title } />
		);

		expect( result ).toMatchSnapshot();
	} );

	test( 'renders pending post as expected', () => {
		const title = {
			rendered: 'Pending Post',
		};

		const result = shallow(
			<RelationshipItem id={ 42 } status="pending" title={ title } />
		);

		expect( result ).toMatchSnapshot();
	} );

	test( 'renders draft post with empty title as expected', () => {
		const title = {
			rendered: '',
		};

		const result = shallow(
			<RelationshipItem id={ 42 } status="draft" title={ title } />
		);

		expect( result ).toMatchSnapshot();
	} );

	test( 'renders post with custom post status and missing title as expected', () => {
		const result = shallow(
			<RelationshipItem id={ 42 } status="other" />
		);

		expect( result ).toMatchSnapshot();
	} );

} );
