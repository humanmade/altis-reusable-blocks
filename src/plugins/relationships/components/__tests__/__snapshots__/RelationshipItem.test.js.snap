// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RelationshipItem/> renders draft post with empty title as expected 1`] = `
<PanelRow>
  <a
    href="https://example.com/edit/42"
    rel="noopener noreferrer"
    target="_blank"
  >
    #42 - (No Title)
  </a>
  (Draft)
</PanelRow>
`;

exports[`<RelationshipItem/> renders pending post as expected 1`] = `
<PanelRow>
  <a
    href="https://example.com/edit/42"
    rel="noopener noreferrer"
    target="_blank"
  >
    #42 - Pending Post
  </a>
  (Pending)
</PanelRow>
`;

exports[`<RelationshipItem/> renders post with custom post status and missing title as expected 1`] = `
<PanelRow>
  <a
    href="https://example.com/edit/42"
    rel="noopener noreferrer"
    target="_blank"
  >
    #42 - (No Title)
  </a>
</PanelRow>
`;

exports[`<RelationshipItem/> renders published post as expected 1`] = `
<PanelRow>
  <a
    href="https://example.com/edit/42"
    rel="noopener noreferrer"
    target="_blank"
  >
    #42 - Published Post
  </a>
</PanelRow>
`;
